yarn-project/aztec-node/src/sentinel/store.ts

```diff
@@ -86,9 +86,11 @@ export class SentinelStore {
     slot: bigint,
     status: 'block-mined' | 'block-proposed' | 'block-missed' | 'attestation-sent' | 'attestation-missed',
   ) {
-    const currentHistory = (await this.getHistory(who)) ?? [];
-    const newHistory = [...currentHistory, { slot, status }].slice(-this.config.historyLength);
-    await this.historyMap.set(who.toString(), this.serializeHistory(newHistory));
+    await this.store.transactionAsync(async () => {
+      const currentHistory = (await this.getHistory(who)) ?? [];
+      const newHistory = [...currentHistory, { slot, status }].slice(-this.config.historyLength);
+      await this.historyMap.set(who.toString(), this.serializeHistory(newHistory));
+    });
```

Comment:
Did you spot an error related to this? Validator status should not be pushed often, it's odd to see a race condition here. Still, I agree that wrapping in a tx is safer. But I'm worried that if you did see an error in prod, it means we're calling the sentinel more than we should.

Reply:
I didn't see any errors, but the first time I ran this I couldn't see any missed slots in the sentinel logs after killing 12/60 validators. I wasn't super thorough to check each and every one of them but still, I would've expected to see some missed proposals/attestations.